Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/intersection.h

```diff
@@ -26,9 +26,769 @@
+  typedef typename CGAL::Box_intersection_d::Box_with_info_d<double, 3, face_descriptor> Box;
+
+  // make one box per facet
+  std::vector<Box> boxes1;
+  std::vector<Box> boxes2;
+  boxes1.reserve(
+        std::distance( boost::begin(face_range1), boost::end(face_range1) )
+        );
+  boxes2.reserve(
+        std::distance( boost::begin(face_range2), boost::end(face_range2) )
+        );
+
+  typedef typename GetVertexPointMap<TM, NamedParameters>::const_type VertexPointMap;
+
+  VertexPointMap vpmap1 = boost::choose_param(get_param(np1, internal_np::vertex_point),
+                                              get_const_property_map(boost::vertex_point, mesh1));
+  VertexPointMap vpmap2 = boost::choose_param(get_param(np2, internal_np::vertex_point),
+                                              get_const_property_map(boost::vertex_point, mesh2));
+
+  BOOST_FOREACH(face_descriptor f, face_range1)
+  {
+    boxes1.push_back(Box(Polygon_mesh_processing::face_bbox(f, mesh1), f));
+  }
+
+  BOOST_FOREACH(face_descriptor f, face_range2)
+  {
+    boxes2.push_back(Box(Polygon_mesh_processing::face_bbox(f, mesh2), f));
+  }
+
+  // generate box pointers
+  std::vector<const Box*> box1_ptr;
+  std::vector<const Box*> box2_ptr;
+  box1_ptr.reserve(num_faces(mesh1));
+  box2_ptr.reserve(num_faces(mesh2));
+
+  BOOST_FOREACH(Box& b, boxes1)
+      box1_ptr.push_back(&b);
+  BOOST_FOREACH(Box& b, boxes2)
+      box2_ptr.push_back(&b);
+
+  // compute intersections filtered out by boxes
+  typedef typename GetGeomTraits<TM, NamedParameters>::type GeomTraits;
+
+  CGAL::internal::Intersect_faces<TM,
+      GeomTraits,
+      Box,
+      OutputIterator,
+      VertexPointMap>
+      Intersect_faces(mesh1, mesh2,
+                       out,
+                       vpmap1, vpmap2,
+                       GeomTraits());
+
+  std::ptrdiff_t cutoff = 2000;
+  CGAL::box_intersection_d(box1_ptr.begin(), box1_ptr.end(),
+                           box2_ptr.begin(), box2_ptr.end(),
+                           Intersect_faces,cutoff);
+  return Intersect_faces.m_iterator;
+}
+
+
+/*
+ * \ingroup PMP_intersection_grp
+ * detects and records intersections between the specified
+ * faces of a triangulated surface mesh and a `Polyline`.
+ * This function depends on the package \ref PkgBoxIntersectionDSummary
+ *
+ * \pre `CGAL::is_triangle_mesh(mesh)`
+ *
+ * \tparam TriangleMesh a model of `FaceListGraph`
+ * \tparam FaceRange range of `boost::graph_traits<TriangleMesh>::%face_descriptor`,
+ *  model of `Range`.
+ * Its iterator type is `RandomAccessIterator`.
+ * \tparam Polyline a `RandomAccessRange` of `CGAL::Point_3`. Those points must come from the same
+ * geometric traits that templates`TriangleMesh`.
+ * \tparam OutputIterator a model of `OutputIterator` holding objects of type
+ *   `std::pair<std::size_t, std::size_t>`. This OutputIterator will hold the position of the
+ *  elements in their respective range.
```

Comment:
It might not be clear for the user that the pair.second is the index in the range of the first point of the segment in the polyline following the order of the range. Also I wonder whether we should have specified that the polyline is considered as a set of segments sharing a vertex and how they are represented. It is obvious to us but maybe not to the user.

Reply:
Isn't it the definition of a polyline ?

Reply:
a polyline is sometime represented as a soup of segments (i.e. endpoints duplicated). Since there is no definition anywhere in CGAL, it is not clear.